frontend/src/pages/add/VendorAdd.tsx

```diff
@@ -0,0 +1,212 @@
+        name: vendor.vendorName,
+        buyback_rate: vendor.buybackRate,
+      };
+      VENDORS_API.addVendor(newVendor);
+    });
+
+    axios
+      .all(vendorRequests)
+      .then((responses) => {
+        responses.forEach((resp) => {
+          showSuccess(toast, "Successfully added vendor");
+        });
+        isGoBackActive ? navigate("/vendors") : window.location.reload();
+      })
+      .catch(() => {
+        showFailure(toast, "One or more of the vendors failed to add");
+      });
+  };
+
+  const actionBodyTemplate = (rowData: VendorRow) => {
+    return (
+      <React.Fragment>
+        <Button
+          type="button"
+          icon="pi pi-trash"
+          className="p-button-rounded p-button-danger"
+          onClick={() => deleteVendor(rowData)}
+        />
+      </React.Fragment>
+    );
+  };
+
+  const addRowButton = (
+    <AddRowButton
+      emptyItem={emptyVendor}
+      rows={vendors}
+      setRows={setVendors}
+      label={"Add Vendor"}
+    />
+  );
+
+  const leftToolbar = <>{addRowButton}</>;
+
+  // Right
+  const submitButton = (
+    <ConfirmPopup
+      isPopupVisible={isConfirmationPopupVisible}
+      hideFunc={() => setIsConfirmationPopupVisible(false)}
+      onFinalSubmission={onSubmit}
+      onShowPopup={() => setIsConfirmationPopupVisible(true)}
+      label={"Submit"}
+      className="p-button-success ml-2"
+    />
+  );
+
+  const submitAndGoBackButton = (
+    <ConfirmPopup
+      isPopupVisible={isConfirmationPopupVisible}
+      hideFunc={() => setIsConfirmationPopupVisible(false)}
+      onFinalSubmission={onSubmit}
+      onRejectFinalSubmission={() => {
+        setIsGoBackActive(false);
+      }}
+      onShowPopup={() => {
+        setIsConfirmationPopupVisible(true);
+        setIsGoBackActive(true);
+      }}
+      label={"Submit and Go Back"}
+      className="p-button-success ml-2"
+    />
+  );
+
+  const rightToolbar = (
+    <>
+      {submitAndGoBackButton}
+      {submitButton}
+    </>
+  );
+
+  const backButton = (
+    <div className="flex col-1">
+      <BackButton onClick={() => navigate("/vendors")} className="ml-1" />
+    </div>
+  );
+
+  const columns = createColumns(COLUMNS);
+
+  return (
+    <div className="grid flex justify-content-center">
+      <Toast ref={toast} />
+      <div className="flex col-12 p-0">
+        {backButton}
+        <div className="pt-2 col-10">
+          <h1 className="p-component p-text-secondary text-5xl text-center text-900 color: var(--surface-800);">
+            Add Vendors
+          </h1>
+        </div>
+      </div>
+      <div className="col-11">
+        <form onSubmit={onSubmit}>
+          <Toolbar className="mb-4" left={leftToolbar} right={rightToolbar} />
+
+          <DataTable
+            showGridlines
+            value={vendors}
+            className="editable-cells-table"
+            responsiveLayout="scroll"
+            editMode="cell"
+          >
+            {columns}
+            <Column
```

Comment:
I actually created a delete column in src/components/datatable/DeleteColumn, you should be able to use that here. Check out the BB/PO/SR page to see how I use it

Reply:
Then should be able to delete "actionBodyTemplate" as well if you use it